test/voice-broadcast/audio/VoiceBroadcastRecorder-test.ts

```diff
@@ -0,0 +1,206 @@
+
+        it("contentType should return the value from VoiceRecording", () => {
+            expect(voiceBroadcastRecording.contentType).toBe(contentType);
+        });
+
+        describe("when the first page from recorder has been received", () => {
+            beforeEach(() => {
+                voiceRecording.onDataAvailable(headers1);
+            });
+
+            itShouldNotEmitAChunkRecordedEvent();
+        });
+
+        describe("when a second page from recorder has been received", () => {
+            beforeEach(() => {
+                voiceRecording.onDataAvailable(headers1);
+                voiceRecording.onDataAvailable(headers2);
+            });
+
+            itShouldNotEmitAChunkRecordedEvent();
+        });
+
+        describe("when a third page from recorder has been received", () => {
+            beforeEach(() => {
+                voiceRecording.onDataAvailable(headers1);
+                voiceRecording.onDataAvailable(headers2);
+                voiceRecording.onDataAvailable(chunk1);
+            });
+
+            itShouldNotEmitAChunkRecordedEvent();
+
+            describe("stop", () => {
+                let stopPayload: ChunkRecordedPayload;
+
+                beforeEach(async () => {
+                    stopPayload = await voiceBroadcastRecording.stop();
+                });
+
+                it("should return the remaining chunk", () => {
+                    expect(stopPayload).toEqual({
+                        buffer: concat(headers1, headers2, chunk1),
+                        length: 23,
+                    });
+                });
+            });
+        });
+
+        describe("when some chunks have been received", () => {
+            beforeEach(() => {
+                // simulate first chunk
+                voiceRecording.onDataAvailable(headers1);
+                voiceRecording.onDataAvailable(headers2);
+                // @ts-ignore
+                voiceRecording.recorderSeconds = 42;
+                voiceRecording.onDataAvailable(chunk1);
+
+                // simulate a second chunk
+                voiceRecording.onDataAvailable(chunk2a);
+                // @ts-ignore
+                voiceRecording.recorderSeconds = 72;
+                voiceRecording.onDataAvailable(chunk2b);
+            });
+
+            it("should emit ChunkRecorded events", () => {
+                expect(voiceRecording.emit).toHaveBeenNthCalledWith(
+                    1,
+                    VoiceBroadcastRecorderEvent.ChunkRecorded,
+                    {
+                        buffer: concat(headers1, headers2, chunk1),
+                        length: 42,
+                    },
+                );
+
+                expect(voiceRecording.emit).toHaveBeenNthCalledWith(
+                    2,
+                    VoiceBroadcastRecorderEvent.ChunkRecorded,
+                    {
+                        buffer: concat(headers1, headers2, chunk2a, chunk2b),
+                        length: 72 - 42,
```

Comment:
these magic numbers need comments :(

Reply:
Agree. One could expect 72 - everything to be zero :upside_down_face: